EU4toV2/Source/V2World/Country/Country.h

```diff
@@ -0,0 +1,166 @@
+#ifndef COUNTRY_H
+#define COUNTRY_H
+
+#include <memory>
+#include <string>
+#include "../../Mappers/CountryDetails/CountryDetails.h"
+#include "../Localisation/Localisation.h"
+#include "../Reforms/UncivReforms.h"
+#include "../Reforms/Reforms.h"
+#include "../Province/Province.h"
+#include "../State/State.h"
+#include "../Leader/Leader.h"
+#include "../Army/Army.h"
+#include "../Diplomacy/Relation.h"
+
+namespace EU4
+{
+	class Country;
+	class World;
+	class Regions;
+}
+namespace mappers
+{
+	class RegimentCostsMapper;
+	class AdjacencyMapper;
+	class LeaderTraitMapper;
+	class PortProvinces;
+	class Unreleasables;
+	class TechSchoolMapper;
+	class CultureMapper;
+	class ReligionMapper;
+	class GovernmentMapper;
+	class CountryMappings;
+	class PartyNameMapper;
+	class PartyTypeMapper;
+	class IdeaEffectMapper;
+	class TechGroupsMapper;
+}
+namespace V2
+{
+	class Relation;
+	class Country
+	{
+	public:
+		Country() = default;
+		Country(
+			const std::string& countriesFileLine, 
+			bool _dynamicCountry, 
+			const mappers::PartyNameMapper& partyNameMapper, 
+			const mappers::PartyTypeMapper& partyTypeMapper);
+		Country(
+			const std::string& _tag, 
+			const std::string& _commonCountryFile, 
+			const mappers::PartyNameMapper& partyNameMapper,
+			const mappers::PartyTypeMapper& partyTypeMapper);
+		void initFromEU4Country(
+			const EU4::Regions& eu4Regions,
+			std::shared_ptr<EU4::Country> _srcCountry,
+			const mappers::TechSchoolMapper& techSchoolMapper,
+			const mappers::CultureMapper& cultureMapper,
+			const mappers::CultureMapper& slaveCultureMapper,
+			const mappers::IdeaEffectMapper& ideaEffectMapper,
+			const mappers::ReligionMapper& religionMapper,
+			const mappers::ProvinceMapper& provinceMapper,
+			const mappers::GovernmentMapper& governmentMapper,
+			const mappers::CountryMappings& countryMapper
+		);
+		void initFromHistory(const mappers::Unreleasables& unreleasablesMapper);
+		void setupPops(
+			double popWeightRatio,
+			int popConversionAlgorithm,
+			const mappers::ProvinceMapper& provinceMapper
+		);
+		void convertArmies(
+			const mappers::RegimentCostsMapper& regimentCostsMapper,
+			const std::map<int, std::shared_ptr<V2::Province>>& allProvinces,
+			const mappers::PortProvinces& portProvincesMapper,
+			const mappers::ProvinceMapper& provinceMapper
+		);
+		
+		void setNewCountry() { newCountry = true; }
+		bool isNewCountry() const { return newCountry; }
+		std::string getTag() const { return tag; }
+		std::shared_ptr<EU4::Country> getSourceCountry() const { return srcCountry; }
```

Comment:
Bloody github doesn't tell me which line exactly you pointed to when you commented. I assume you're speaking of getSourceCountry that returns the pointer. 

I agree. I wanted to redesign this so that EU4country deposits all relevant data on init, and then leaves for good. However, it's a shitload of data and one isn't certain how much is needed until all of the converter's reassessed. In the next iteration I'll take it out. In the current iteration I managed to take out V2World* which was also sometimes used without need.

Reply:
Yeah, it takes some doing. As long as you have it as a target.